ui/apps/dashboard/src/utils/time.ts

```diff
@@ -0,0 +1,9 @@
+import dayjs from 'dayjs';
+
+export function calculateDuration(start: string | undefined) {
+  if (!start) return '-';
+  const startDate = dayjs(start);
+  const endDate = dayjs(); // 当前时间
```

Comment:
The backend usually returns the server time, which should be UTC by default, and it may be necessary to handle the time zone differences; here, is it the browser time being obtained?

Reply:
it's a good question 🤔 , and dayjs(which we used in our project) use browser timezone in default, and if the timezone differs from local(here refer our personal computer) and server(which means the karmada-dashboard-api ), the calculation of duration may be wrong

Reply:
dayjs support plugin utc and timezone.

Reply:
> dayjs support plugin utc and timezone.

yeah, but that's not key problem, key problem is the difference between client time for different regions. In fact the duration for workload is an absolute number, should be different across regions.

Reply:
maybe just convert it to the current time zone of the browser.

No need to consider multiple time zones. like this.

```typescript
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

export function calculateDuration(start: string | undefined) {
  if (!start) return '-';
  // Convert UTC server time to browser's local timezone
  const startDate = dayjs.utc(start).local();
  const endDate = dayjs();
  const duration = dayjs.duration(endDate.diff(startDate));
  return duration.humanize();
}
```